test/nodes/test_guardrails.py

```
@@ -0,0 +1,938 @@
+            profile_name = field_def['object']
+            if profile_name not in profiles:
+                continue
+            for prop_id in field_def.get('properties', []):
+                # The prop_id should be a valid field key OR match a preconfig key
+                assert prop_id in fields, f'Property "{prop_id}" referenced by "{field_id}" is not defined in fields'
+
+    def test_custom_profile_exposes_all_engine_knobs(self):
+        """The custom profile should include every configurable engine parameter."""
+        services = self._load_services_json()
+        custom_profile = services['preconfig']['profiles']['custom']
+        custom_keys = {k for k in custom_profile.keys() if k != 'title'}
```

Ruff (0.15.7)

[warning] Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)